include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge.h

```diff
@@ -166,34 +272,408 @@ struct __parallel_merge_submitter<_IdType, __internal::__optional_kernel_name<_N
+           |  rng[2]           |   +   |       |       |       |     |                        | slm[1]     |                                        | SLM bank: write into one SLM bank from one work-item
+           |  rng[3]           |   +   |       |       |       |     |                        | slm[2]     |                                       /
+           |  rng[4]           |       |   +   |       |       |     |                        | slm[3]     |
+           |  rng[5]           |       |   +   |       |       |     |                        | slm[4]     |
+           |  rng[6]           |       |   +   |       |       |     |                        | slm[5]     |
+           |  rng[7]           |       |       |   +   |       |     |                        | slm[6]     |
+           |  rng[8]           |       |       |   +   |       |     |                        | slm[7]     |
+           |  rng[9]           |       |       |   +   |       |     |                        | slm[8]     |
+           |  rng[10]          |       |       |       |   +   |     |                        | slm[9]     |
+           |  rng[11]          |       |       |       |   +   |     |                        | slm[10]    |
+           |  rng[12]          |       |       |       |   +   |     |                        | slm[11]    |
+           |  .....            |       |       |       |       | +++ |                        | ...        |  
+           |  rng[M + 1]       |       |       |       |       |     |           +            | slm[M]     |  
+           |  rng[M + 2]       |       |       |       |       |     |           +            | slm[M + 1] |  
+           |  rng[M + 3]       |       |       |       |       |     |           +            | slm[M + 2] |  
+           |  rng[M + 4]       |       |       |       |       |     |                        | slm[M + 3] |
+           |  rng[M + 5]       |       |       |       |   -   |     |                        |            |  <--- __idx_global_end
+           |  .....            |       |       |       |       | --- |                        |            |  
+           |  rng[M + M + 1]   |       |       |       |       |     |           -            |            |  
+           +-------------------+--------------------------------------------------------------+------------+
+                                                   ^
+                                                   |
+                                              __local_id
+           
+            "+" - load one source data item ito SLM
+        */
+
+        const std::size_t __wg_data_size_rng = __idx_global_end - __idx_global_begin;
+        if (__wg_data_size_rng > 0)
+        {
+            // Calculate the size of the current part of merging data per work-item
+            const std::size_t __loading_data_per_wi = oneapi::dpl::__internal::__dpl_ceiling_div(__rng.size(), __wi_in_one_wg);
```

Comment:
Could you clarify this for me if I am not following the code correctly? `__rng` looks to be the input buffer, so `__loading_data_per_wi` will be very large for large input. Is this correct?